File: Model/PortefeuilleElectronique.js
class PortefeuilleElectronique {
  constructor(soldeInitial = 0, categorie = "Général") {
    this.solde = soldeInitial;
    this.categorie = categorie;
  }

  getSolde() {
    return this.solde;
  }

  getCategorie() {
    return this.categorie;
  }

  deposer(montant) {
    if (montant > 0) {
      this.solde += montant;
      return true;
    } else {
      return false;
    }
  }

  retirer(montant) {
    if (montant > 0 && montant <= this.solde) {
      this.solde -= montant;
      return true;
    } else {
      return false;
    }
  }
}
module.exports = PortefeuilleElectronique;

File: Model/PortefeuilleElectronique.py
class PortefeuilleElectronique:
    def __init__(self, solde_initial=0, categorie="Général"):
        self.solde = solde_initial
        self.categorie = categorie

    def get_solde(self):
        return self.solde

    def deposer(self, montant):
        if montant > 0:
            self.solde += montant
            return True
        else:
            return False

    def retirer(self, montant):
        if 0 < montant <= self.solde:
            self.solde -= montant
            return True
        else:
            return False